src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,163 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public void add(T value) {
+        if (head == null) {
+            head = new Node<>(value, null, null);
+            tail = head;
+        } else {
+            Node<T> newNode = new Node<>(value, tail, null);
+            tail.next = newNode;
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == 0 && head == null) {
+            add(value);
+        } else if (index == 0 && head != null) {
+            Node<T> newNode = new Node<>(value, null, head);
+            head.prev = newNode;
+            head = newNode;
+            size++;
+        } else {
+            Node<T> previousNode = iteratorByIndex(index - 1);
+
+            if (previousNode.next == null) {
+                Node<T> newNode = new Node<>(value, previousNode, null);
+                previousNode.next = newNode;
+                previousNode = newNode;
+                size++;
+            } else {
+                Node<T> newNode = new Node<>(value, previousNode, previousNode.next);
+                previousNode.next = newNode;
+                newNode.next.prev = newNode;
+                size++;
+            }
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return iteratorByIndex(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        T oldValue = iteratorByIndex(index).value;
+        iteratorByIndex(index).value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> removedNode = iteratorByIndex(index);
+        removeNode(removedNode);
+        return removedNode.value;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        Node<T> removedNode = iteratorByValue(object);
+        if (removedNode == null) {
+            return false;
+        }
+        removeNode(removedNode);
+        return true;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private Node<T> iteratorByIndex(int index) {
+        if (isEmpty() || index >= size || index < 0) {
+            throw new IndexOutOfBoundsException("Index [" + index + "] out of bounds!");
+        }
```

Comment:
make a new method to check index and use it in another method (like remove or add by index)